storage/pebble/register_storage.go

```diff
@@ -178,10 +178,12 @@ func newLookupKey(height uint64, key []byte) *lookupKey {
 	return &lookupKey
 }
 
-// GetSnapshotAt returns a snapshot of the register index at the given block height.
-// the snapshot has a cache. Nil values are cached.
+// GetSnapshotAt returns a snapshot of the register index at the start of the
+// given block height (which is the end of the previous block).
+// The snapshot has a cache. Nil values are cached.
 func (r *RegisterStorage) GetSnapshotAt(evmBlockHeight uint64) (types.BackendStorageSnapshot, error) {
-	return NewStorageSnapshot(r.Get, evmBlockHeight), nil
+	// `evmBlockHeight-1` to get the end state of the previous block.
+	return NewStorageSnapshot(r.Get, evmBlockHeight-1), nil
```

Comment:
Can we let the caller do the `-1` and deals with potential underflow? So the `GetSnapshotAt` always return the end state. I think end state is easy to think.

Reply:
the [interface](https://github.com/onflow/flow-go/issues/6962#issuecomment-2627790608) defines it as the start of block state

Reply:
+1 for handling the underflow. it would only happen on block 0, but that happens when syncing from the start or when starting the emulator.

Reply:
Added in https://github.com/onflow/flow-evm-gateway/pull/760/commits/ea6e8338f9b2c51cb926a775c7770922a33e2201